pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -99,26 +99,33 @@ func NewController(client kube.Client, c model.ConfigStoreCache, options control
 			}
 		}, uint(features.StatusMaxWorkers))
 	}
-	nsInformer := client.KubeInformer().Core().V1().Namespaces().Informer()
+	var nsLister listerv1.NamespaceLister
+	var nsInformer cache.SharedIndexInformer
+	if client.GetMemberRoll() == nil {
+		nsInformer = client.KubeInformer().Core().V1().Namespaces().Informer()
+		nsLister = client.KubeInformer().Core().V1().Namespaces().Lister()
+	}
 	gatewayController := &Controller{
 		client:            client,
 		cache:             c,
-		namespaceLister:   client.KubeInformer().Core().V1().Namespaces().Lister(),
+		namespaceLister:   nsLister,
 		namespaceInformer: nsInformer,
 		domain:            options.DomainSuffix,
 		status:            statusQueue,
 		// Disabled by default, we will enable only if we win the leader election
 		statusEnabled: atomic.NewBool(false),
 	}
 
-	nsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			gatewayController.namespaceEvent(nil, obj)
-		},
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			gatewayController.namespaceEvent(oldObj, newObj)
-		},
-	})
+	if client.GetMemberRoll() == nil {
```

Comment:
nit: I'd prefer a comparison like `if nsInformer != nil` here and below, but it's not a strong preference.

Reply:
makes sense. I made the changes